### ISSUE TITLE
Strange itk.imread behavior with fallback_only=True

### ISSUE THREAD

**Issue by jond01** [InsightSoftwareConsortium/ITK#1835]
### Description

`itk.imread`'s parameter `fallback_only=True` does not accept runtime errors.
https://github.com/InsightSoftwareConsortium/ITK/blob/7900a8b6d63ec8201405eff3b46c8d2a4e563c45/Wrapping/Generators/Python/itkExtras.py#L663

### Steps to Reproduce + behavior
Download the dicom file:
![CT0012](https://user-images.githubusercontent.com/36337649/82836729-ccd2b000-9ecf-11ea-9202-4dec91ca9b38.jpg)
(Please remove the '.jpg' suffix, it was added for github)

Cases:
1. `itk.imread('path/to/CT0012')` -> doesn't work
2. `itk.imread('path/to/CT0012', pixel_type=itk.SS, fallback_only=True)` -> doesn't work
3. `itk.imread('path/to/CT0012', pixel_type=itk.SS)` -> works

### Expected behavior

I would expect 2 to work if 3 works.

### Actual behavior

The (super long) runtime error for 1 and 2 is:
```python
>>> itk.imread('path/to/CT0012')
Traceback (most recent call last):
  File "...\lib\site-packages\itkTemplate.py", line 334, in __getitem__
    return(self.__template__[tuple(cleanParameters)])
KeyError: (<itkCType signed int>, 3)
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "...\lib\site-packages\itkTemplate.py", line 338, in __getitem__
    return(self.__template__[tuple(cleanParameters)])
KeyError: (<itkCType signed int>, 3)
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "...\lib\site-packages\itkExtras.py", line 572, in imread
    reader = TemplateReaderType.New(**kwargs)
  File "...\lib\site-packages\itkTemplate.py", line 471, in New
    return self._NewImageReader(itk.ImageFileReader, False, 'FileName', *args, **kwargs)
  File "...\lib\site-packages\itkTemplate.py", line 555, in _NewImageReader
    ImageType = itk.Image[PixelType, dimension]
  File "...\lib\site-packages\itkTemplate.py", line 340, in __getitem__
    raise TemplateTypeError(self, tuple(cleanParameters))
itkTemplate.TemplateTypeError: itk.Image is not wrapped for input type `itk.SI, int`.
To limit the size of the package, only a limited number of
types are available in ITK Python. To print the supported
types, run the following command in your python environment:
    itk.Image.GetTypes()
Possible solutions:
* If you are an application user:
** Convert your input image into a supported format (see below).
** Contact developer to report the issue.
* If you are an application developer, force input images to be
loaded in a supported pixel type.
    e.g.: instance = itk.Image[itk.RGBPixel[itk.UC], int].New(my_input)
* (Advanced) If you are an application developer, build ITK Python yourself and
turned to `ON` the corresponding CMake option to wrap the pixel type or image
dimension you need. When configuring ITK with CMake, you can set
`ITK_WRAP_${type}` (replace ${type} with appropriate pixel type such as
`double`). If you need to support images with 4 or 5 dimensions, you can add
these dimensions to the list of dimensions in the CMake variable
`ITK_WRAP_IMAGE_DIMS`.
Supported input types:
itk.RGBPixel[itk.UC]
itk.RGBAPixel[itk.UC]
itk.Vector[itk.F,2]
itk.Vector[itk.F,3]
itk.Vector[itk.F,4]
itk.CovariantVector[itk.F,2]
itk.CovariantVector[itk.F,3]
itk.CovariantVector[itk.F,4]
itk.SS
itk.UC
itk.US
itk.F
itk.complex[itk.F]
itk.Vector[itk.D,2]
itk.Vector[itk.D,3]
itk.Vector[itk.D,4]
itk.D
itk.UL
itk.ULL
itk.B
itk.FixedArray[itk.F,2]
itk.Offset[2]
itk.SymmetricSecondRankTensor[itk.D,2]
itk.RGBPixel[itk.UC]
itk.RGBAPixel[itk.UC]
itk.Vector[itk.F,2]
itk.Vector[itk.F,3]
itk.Vector[itk.F,4]
itk.CovariantVector[itk.F,2]
itk.CovariantVector[itk.F,3]
itk.CovariantVector[itk.F,4]
itk.SS
itk.UC
itk.US
itk.F
itk.complex[itk.F]
itk.Vector[itk.D,2]
itk.Vector[itk.D,3]
itk.Vector[itk.D,4]
itk.D
itk.UL
itk.ULL
itk.B
itk.FixedArray[itk.F,3]
itk.Offset[3]
itk.SymmetricSecondRankTensor[itk.D,3]
itk.Vector[itk.F,1]
itk.Vector[itk.D,2]
itk.Vector[itk.D,3]
itk.Vector[itk.D,4]
itk.Vector[itk.F,1]
itk.CovariantVector[itk.D,2]
itk.CovariantVector[itk.D,3]
itk.CovariantVector[itk.D,4]
itk.CovariantVector[itk.D,2]
itk.CovariantVector[itk.D,3]
itk.CovariantVector[itk.D,4]
itk.NormalBandNode[itk.Image[itk.F,2]]
itk.NormalBandNode[itk.Image[itk.F,3]]
```

### Reproducibility

From my experience, it occurs mainly for multiframe dicoms.

### Additional Information
https://github.com/InsightSoftwareConsortium/ITK/blob/7900a8b6d63ec8201405eff3b46c8d2a4e563c45/Wrapping/Generators/Python/itkExtras.py#L678-L684
As we can see, `fallback_only` is relevant only for key-errors (such as passing `pixel_type=55`).

This can be easily solved with `except (KeyError, RuntimeError):` in L683. However, allowing every runtime-error might be too permissive.

What do you think?

**Comment by thewtex** [comment]
@jond01 good catch.

> This can be easily solved with except (KeyError, RuntimeError): in L683. However, allowing every runtime-error might be too permissive.

Good idea.

To be more specific, perhaps we could check the error message or introduce a new derived error type.

**Comment by jond01** [comment]
Oops, I'm taking back my words about the runtime-error.
It's actually an error of type `itkTemplate.TemplateTypeError`:

>  File "...\lib\site-packages\itkTemplate.py", line 340, in __getitem__
  raise TemplateTypeError(self, tuple(cleanParameters))
  itkTemplate.TemplateTypeError: itk.Image is not wrapped for input type `itk.SI, int`.
  To limit the size of the package, only a limited number of

Which is similar in its nature to a KeyError:
https://github.com/InsightSoftwareConsortium/ITK/blob/9a5a00fc72d4ed13ab4a8cbe2a26e4c6b104732f/Wrapping/Generators/Python/itkTemplate.py#L517-L524

To summarize, the situation is better because we can add to L683:
```python
except (KeyError, TemplateTypeError):
```

Alternatively, we can make TemplateTypeError inherit also from KeyError:
Change
https://github.com/InsightSoftwareConsortium/ITK/blob/9a5a00fc72d4ed13ab4a8cbe2a26e4c6b104732f/Wrapping/Generators/Python/itkTemplate.py#L55
To
```python
class TemplateTypeError(TypeError, KeyError):
```
